crates/noirc_driver/src/lib.rs

```diff
@@ -67,8 +68,15 @@ impl Default for CompileOptions {
 
 impl Driver {
     pub fn new(language: &Language, is_opcode_supported: Box<dyn Fn(&Opcode) -> bool>) -> Self {
-        let mut driver = Driver { context: Context::default(), language: language.clone() };
-        driver.context.def_interner.set_opcode_support(is_opcode_supported);
+        let mut driver =
+            Driver { context: Context::default(), language: language.clone(), is_opcode_supported };
+
+        // We cannot pass in the real version of `is_opcode_supported` here as we cannot clone the boxed closure.
+        // TODO(#1102): remove the requirement for the `NodeInterner` to know about which opcodes the backend supports.
```

Comment:
Currently the only way to avoid this is to either:

- pass in two (potentially more in the future) separate closures.
- store an entire instance of the backend inside `Driver`